### ISSUE TITLE
NullReferenceException in TextMateIndentationTextEditorExtension.cs:58 (KeyPress)

### ISSUE THREAD

**Issue by KirillOsenkov** [mono/monodevelop#5640]
@mkrueger I have a customer log with this nullref, do you think this has been fixed already?

If not, any ideas what could be null in that area? Thanks!

```
ERROR [2018-07-25 19:15:38Z]: An unhandled exception has occured. Terminating Visual Studio? False
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Ide.Editor.Extension.ErrorHandlerTextEditorExtension+<>c__DisplayClass5_0+<<DocumentContext_DocumentParsed>b__0>d.MoveNext () [0x0003b] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/ErrorHandlerTextEditorExtension.cs:85 
ERROR [2018-07-25 19:15:49Z]: Error in text editor extension chain
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Ide.Editor.TextMate.TextMateIndentationTextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.TextMate/TextMateIndentationTextEditorExtension.cs:58 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.AutoInsertBracketTextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/AutoInsertBracketTextEditorExtension.cs:61 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Debugger.ExceptionCaughtTextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x0005d] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ExceptionCaughtDialog.cs:1106 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.CompletionTextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00108] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/CompletionTextEditorExtension.cs:164 
  at WebToolingAddin.RoslynCompletionEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in C:\BuildAgent\work\feeffd8e4406e757\XP\src\Editors\WebToolingAddin\VSEditor\Extensions\RoslynCompletionEditorExtension.cs:158 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at WebToolingAddin.BraceCompletionEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00174] in C:\BuildAgent\work\feeffd8e4406e757\XP\src\Editors\WebToolingAddin\VSEditor\Braces\BraceCompletionEditorExtension.cs:123 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at WebToolingAddin.MonoDevelopCommandTarget.Invoke (System.Guid group, System.Int32 id, System.Object inputArg, System.Object& outputArg) [0x0000b] in C:\BuildAgent\work\feeffd8e4406e757\XP\src\Editors\WebToolingAddin\WebTooling\Web\Extensions\MonoDevelopCommandTarget.cs:25 
  at Microsoft.Web.Editor.Controller.Controller.Invoke (System.Guid group, System.Int32 id, System.Object inputArg, System.Object& outputArg) [0x0002f] in C:\BuildAgent\work\feeffd8e4406e757\XP\src\Editors\Web\Editor\Impl\Controller\Controller.cs:56 
  at Microsoft.Web.Editor.Controller.ViewController.Invoke (System.Guid group, System.Int32 id, System.Object inputArg, System.Object& outputArg) [0x00078] in C:\BuildAgent\work\feeffd8e4406e757\XP\src\Editors\Web\Editor\Impl\Controller\ViewController.cs:132 
  at Microsoft.Html.Editor.Commands.HtmlMainController.Invoke (System.Guid group, System.Int32 id, System.Object inputArg, System.Object& outputArg) [0x000ce] in C:\BuildAgent\work\feeffd8e4406e757\XP\src\Editors\html\Editor\Impl\Commands\HtmlMainController.cs:144 
  at WebToolingAddin.WebEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00016] in C:\BuildAgent\work\feeffd8e4406e757\XP\src\Editors\WebToolingAddin\WebTooling\Web\Extensions\WebEditorExtension.cs:106 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.SourceEditor.ExtensibleTextEditor.ExtensionKeyPress (Gdk.Key key, System.UInt32 ch, Gdk.ModifierType state) [0x0003a] in /Users/builder/data/lanes/5965/498923ea/source/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/ExtensibleTextEditor.cs:282 

```

> VS bug [#661694](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/661694)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.126 (release-7.7)

Pull Request #5647 merged by: Mike Krüger
Author: mono
Commit: aa41b42b0b51e7b55165d4dfff667340686459f4 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.931 (release-7.7)

Pull Request #5647 merged by: Mike Krüger
Author: mono
Commit: aa41b42b0b51e7b55165d4dfff667340686459f4 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5647 merged by: Mike Kr??ger
Author: mono
Commit: aa41b42b0b51e7b55165d4dfff667340686459f4 (xamarin/monodevelop)
